Entities/MOSRotating.cpp

```diff
@@ -1531,12 +1530,8 @@ bool MOSRotating::DrawMOIDIfOverlapping(MovableObject *pOverlapMO)
         float combinedRadii = GetRadius() + pOverlapMO->GetRadius();
         Vector otherPos = pOverlapMO->GetPos();
 
-        // Quick check
-        if (fabs(otherPos.m_X - m_Pos.m_X) > combinedRadii || fabs(otherPos.m_Y - m_Pos.m_Y) > combinedRadii)
```

Comment:
That code was an AABB early-out optimisation to avoid the sqrt call. With this PR to compare squared distances, that code is no longer an optimisation and actually slows things down.